File: a3/srcbackend/main/java/com/utcn/demo/entity/Question.java
package com.utcn.demo.entity;


import jakarta.persistence.*;

@Entity
@Table(name = "question")
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="id")
    private Long questionId;

    @Column(name="title")
    private String title;


    @OneToOne
    @JoinColumn(name="c_id")
    private Content questionContent;

    public Question(){

    }

    public Question( String title, Content content) {

        this.title = title;
        this.questionContent = content;
    }

    public Long getQuestionId() {
        return questionId;
    }

    public void setQuestionId(Long questionId) {
        this.questionId = questionId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public Content getQuestionContent() {
        return questionContent;
    }

    public void setQuestionContent(Content content) {
        this.questionContent = content;
    }
}

File: a3/appfrontend/pages/question/question.ts
import {Content} from "../content";

export class Question {



  private title:string;

  private questionContent:Content;


  constructor( title: string, questionContent: Content) {

    this.title = title;
    this.questionContent = questionContent;
  }





  get _title(): string {
    return this.title;
  }

  set _title(value: string) {
    this.title = value;
  }

  get _questionContent(): Content {
    return this.questionContent;
  }

  set _questionContent(value: Content) {
    this.questionContent = value;
  }
}
